Header header
bool if_collision
